include/ddc/kernels/fft.hpp

```diff
@@ -357,224 +136,127 @@ double b(ddc::DiscreteDomain<DDimX...> x_mesh)
            / 2 / (N<DDim>(x_mesh) - 1);
 }
 
-// core
-template <typename Tin, typename Tout, typename ExecSpace, typename MemorySpace, typename... DDimX>
-void core(
-        ExecSpace const& execSpace,
-        Tout* out_data,
-        Tin* in_data,
-        ddc::DiscreteDomain<DDimX...> mesh,
-        const kwArgs_core& kwargs)
+template <typename... DDimX>
+static constexpr KokkosFFT::axis_type<sizeof...(DDimX)> axes()
+{
+    return KokkosFFT::axis_type<sizeof...(DDimX)> {
+            static_cast<int>(ddc::type_seq_rank_v<DDimX, ddc::detail::TypeSeq<DDimX...>> + 1)...};
+}
+
+KokkosFFT::Normalization ddc_fft_normalization_to_kokkos_fft(
+        FFT_Normalization const ddc_fft_normalization)
+{
+    KokkosFFT::Normalization kokkos_fft_normalization;
+    switch (ddc_fft_normalization) {
+    case ddc::FFT_Normalization::OFF:
+        kokkos_fft_normalization = KokkosFFT::Normalization::none;
+        break;
+    case ddc::FFT_Normalization::FORWARD:
+        kokkos_fft_normalization = KokkosFFT::Normalization::forward;
+        break;
+    case ddc::FFT_Normalization::BACKWARD:
+        kokkos_fft_normalization = KokkosFFT::Normalization::backward;
+        break;
+    case ddc::FFT_Normalization::ORTHO:
+        kokkos_fft_normalization = KokkosFFT::Normalization::ortho;
+        break;
+    // Last case is FULL which is mesh-dependant and thus handled by DDC.
+    default:
+        kokkos_fft_normalization = KokkosFFT::Normalization::none;
+    }
+
+    return kokkos_fft_normalization;
+};
+
+// impl
+template <
+        typename Tin,
+        typename Tout,
+        typename ExecSpace,
+        typename MemorySpace,
+        typename layout_in,
+        typename layout_out,
+        typename DomainOut,
+        typename... DDimIn>
```

Comment:
I would have prefered to keep things symmetric and template with `typename... DDimIn, typename... DDimOut`. But it seems there is a GPU limitation (that could be very annoying when going for a batched FFT):

![image](https://github.com/user-attachments/assets/a1445005-4763-4731-9783-9857dfdb7595)